zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java

```diff
@@ -192,34 +192,34 @@ public boolean accept(Path entry) throws IOException {
          *    {ZEPPELIN_HOME}/interpreter/{interpreter_name}
          * 3. Register it by Interpreter.register
          */
-        if (!registerInterpreterFromPath(interpreterDirString, interpreterJson)) {
-          if (!registerInterpreterFromResource(cl, interpreterDirString, interpreterJson)) {
-            /*
-             * TODO(jongyoul)
-             * - Remove these codes below because of legacy code
-             * - Support ThreadInterpreter
-            */
-            URLClassLoader ccl = new URLClassLoader(
-                    recursiveBuildLibList(interpreterDir.toFile()), cl);
-            for (String className : interpreterClassList) {
-              try {
-                // Load classes
-                Class.forName(className, true, ccl);
-                Set<String> interpreterKeys = Interpreter.registeredInterpreters.keySet();
-                for (String interpreterKey : interpreterKeys) {
-                  if (className
-                      .equals(Interpreter.registeredInterpreters.get(interpreterKey)
-                          .getClassName())) {
-                    Interpreter.registeredInterpreters.get(interpreterKey)
-                        .setPath(interpreterDirString);
-                    logger.info("Interpreter " + interpreterKey + " found. class=" + className);
-                    cleanCl.put(interpreterDirString, ccl);
-                  }
-                }
-              } catch (Throwable t) {
-                // nothing to do
+        registerInterpreterFromPath(interpreterDirString, interpreterJson);
+        registerInterpreterFromResource(cl, interpreterDirString, interpreterJson);
```

Comment:
This would cause interpreter to be registered twice which make it difficult to configure the default interpreter setting. I change the method signature to return boolean to indicate whether it is registered successfully. 
See my comments.
```
         * Register interpreter by the following ordering
         * 1. Register it from path {ZEPPELIN_HOME}/interpreter/{interpreter_name}/
         *    interpreter-setting.json
         * 2. Register it from interpreter-setting.json in classpath
         *    {ZEPPELIN_HOME}/interpreter/{interpreter_name}
         * 3. Register it by Interpreter.register
```